src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/custom.py

```diff
@@ -82,22 +82,30 @@ def iot_hub_sku_list(client, hub_name, resource_group_name=None):
     return client.get_valid_skus(resource_group_name, hub_name)
 
 
-def iot_hub_consumer_group_create(client, hub_name, consumer_group_name, resource_group_name=None, event_hub_name='events'):
+def iot_hub_consumer_group_create(client, hub_name, consumer_group_name, resource_group_name=None, event_hub_name=None):
+    if event_hub_name is None:
```

Comment:
@derekbekoe The reason I am changing this is because after adding support for `--ids` parameter, error will be thrown for optional parameter with valid default values. You can refer to below errors.
Since I am not very familiar with the code for `--ids` parameters, I changed the default value to None as a work-around.
Do you have any suggestion on how to fix this error? 
Thanks!

```bash
>az iot hub consumer-group list --ids /subscriptions/faab228d-df7a-4086-991ee81c4659d41a/resourceGroups/ao-iot-sample/providers/Microsoft.Devices/IotHubs/ao-iot-hub/eventHubEndpoints/events

'str' object has no attribute 'append'
Traceback (most recent call last):
  File "d:\workspace\azure-cli-x\src\azure-cli-core\azure\cli\core\commands\arm.py", line 120, in __call__
    existing_values.append(parts[arg.id_part])
AttributeError: 'str' object has no attribute 'append'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "d:\workspace\azure-cli-x\src\azure-cli\azure\cli\main.py", line 33, in main
    cmd_result = APPLICATION.execute(args)
  File "d:\workspace\azure-cli-x\src\azure-cli-core\azure\cli\core\application.py", line 112, in execute
    args = self.parser.parse_args(argv)
  File "C:\Python35\lib\argparse.py", line 1726, in parse_args
    args, argv = self.parse_known_args(args, namespace)
  File "C:\Python35\lib\argparse.py", line 1758, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Python35\lib\argparse.py", line 1946, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "C:\Python35\lib\argparse.py", line 1923, in consume_positionals
    take_action(action, args)
  File "C:\Python35\lib\argparse.py", line 1832, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\Python35\lib\argparse.py", line 1129, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "C:\Python35\lib\argparse.py", line 1758, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Python35\lib\argparse.py", line 1946, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "C:\Python35\lib\argparse.py", line 1923, in consume_positionals
    take_action(action, args)
  File "C:\Python35\lib\argparse.py", line 1832, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\Python35\lib\argparse.py", line 1129, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "C:\Python35\lib\argparse.py", line 1758, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Python35\lib\argparse.py", line 1946, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "C:\Python35\lib\argparse.py", line 1923, in consume_positionals
    take_action(action, args)
  File "C:\Python35\lib\argparse.py", line 1832, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\Python35\lib\argparse.py", line 1129, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "C:\Python35\lib\argparse.py", line 1758, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Python35\lib\argparse.py", line 1946, in _parse_known_args
    positionals_end_index = consume_positionals(start_index)
  File "C:\Python35\lib\argparse.py", line 1923, in consume_positionals
    take_action(action, args)
  File "C:\Python35\lib\argparse.py", line 1832, in take_action
    action(self, namespace, argument_values, option_string)
  File "C:\Python35\lib\argparse.py", line 1129, in __call__
    subnamespace, arg_strings = parser.parse_known_args(arg_strings, None)
  File "C:\Python35\lib\argparse.py", line 1758, in parse_known_args
    namespace, args = self._parse_known_args(args, namespace)
  File "C:\Python35\lib\argparse.py", line 1964, in _parse_known_args
    start_index = consume_optional(start_index)
  File "C:\Python35\lib\argparse.py", line 1904, in consume_optional
    take_action(action, args, option_string)
  File "C:\Python35\lib\argparse.py", line 1832, in take_action
    action(self, namespace, argument_values, option_string)
  File "d:\workspace\azure-cli-x\src\azure-cli-core\azure\cli\core\commands\arm.py", line 123, in __call__
    raise ValueError(ex)
ValueError: 'str' object has no attribute 'append'
```